server/schedulers/hot_region.go

```diff
@@ -237,6 +237,10 @@ func (h *balanceHotRegionsScheduler) calcScore(items []*core.RegionStat, cluster
 }
 
 func (h *balanceHotRegionsScheduler) balanceByPeer(cluster schedule.Cluster, storesStat core.StoreHotRegionsStat) (*core.RegionInfo, *metapb.Peer, *metapb.Peer) {
+	if h.limiter.OperatorCount(schedule.OpRegion) >= cluster.GetRegionScheduleLimit() {
```

Comment:
Maybe put this into a function just like `allowBalancePeer` in `balanceAdjacentRegionScheduler`?

Reply:
good idea.